src/fat.rs

```diff
@@ -204,106 +206,100 @@ impl<'c, H: HugrView> FatNode<'c, CFG, H> {
+impl<OT, H> PartialEq<Node> for FatNode<'_, OT, H> {
     fn eq(&self, other: &Node) -> bool {
         &self.node == other
     }
 }
 
-impl<'c, OT, H> PartialEq<FatNode<'c, OT, H>> for Node {
-    fn eq(&self, other: &FatNode<'c, OT, H>) -> bool {
+impl<OT, H> PartialEq<FatNode<'_, OT, H>> for Node {
+    fn eq(&self, other: &FatNode<'_, OT, H>) -> bool {
         self == &other.node
     }
 }
 
-impl<'c, 'd, OT1, OT2, H1, H2> PartialEq<FatNode<'d, OT1, H1>> for FatNode<'c, OT2, H2> {
-    fn eq(&self, other: &FatNode<'d, OT1, H1>) -> bool {
+impl<OT1, OT2, H1, H2> PartialEq<FatNode<'_, OT1, H1>> for FatNode<'_, OT2, H2> {
+    fn eq(&self, other: &FatNode<'_, OT1, H1>) -> bool {
         self.node == other.node
     }
 }
 
-impl<'c, OT, H> Eq for FatNode<'c, OT, H> {}
+impl<OT, H> Eq for FatNode<'_, OT, H> {}
 
-impl<'c, OT, H> PartialOrd<Node> for FatNode<'c, OT, H> {
+impl<OT, H> PartialOrd<Node> for FatNode<'_, OT, H> {
     fn partial_cmp(&self, other: &Node) -> Option<Ordering> {
         self.node.partial_cmp(other)
     }
 }
 
-impl<'c, OT, H> PartialOrd<FatNode<'c, OT, H>> for Node {
-    fn partial_cmp(&self, other: &FatNode<'c, OT, H>) -> Option<Ordering> {
+impl<OT, H> PartialOrd<FatNode<'_, OT, H>> for Node {
+    fn partial_cmp(&self, other: &FatNode<'_, OT, H>) -> Option<Ordering> {
         self.partial_cmp(&other.node)
     }
 }
 
-impl<'c, 'd, OT1, OT2, H1, H2> PartialOrd<FatNode<'d, OT1, H1>> for FatNode<'c, OT2, H2> {
-    fn partial_cmp(&self, other: &FatNode<'d, OT1, H1>) -> Option<Ordering> {
+impl<OT1, OT2, H1, H2> PartialOrd<FatNode<'_, OT1, H1>> for FatNode<'_, OT2, H2> {
+    fn partial_cmp(&self, other: &FatNode<'_, OT1, H1>) -> Option<Ordering> {
         self.partial_cmp(&other.node)
     }
 }
 
-impl<'c, OT, H> Ord for FatNode<'c, OT, H> {
+impl<OT, H> Ord for FatNode<'_, OT, H> {
     fn cmp(&self, other: &Self) -> Ordering {
         self.node.cmp(&other.node)
     }
 }
 
-impl<'c, OT, H> Hash for FatNode<'c, OT, H> {
+impl<OT, H> Hash for FatNode<'_, OT, H> {
     fn hash<HA: std::hash::Hasher>(&self, state: &mut HA) {
         self.node.hash(state);
     }
 }
 
-impl<'c, OT: 'c, H: HugrView + ?Sized> AsRef<OT> for FatNode<'c, OT, H>
+impl<OT, H: HugrView + ?Sized> AsRef<OT> for FatNode<'_, OT, H>
 where
     for<'a> &'a OpType: TryInto<&'a OT>,
 {
     fn as_ref(&self) -> &OT {
-        self.get()
+        self.hugr.get_optype(self.node).try_into().ok().unwrap()
     }
 }
 
-impl<'c, OT, H: HugrView + ?Sized> Deref for FatNode<'c, OT, H>
+impl<OT, H: HugrView + ?Sized> Deref for FatNode<'_, OT, H>
 where
-    &'c OpType: TryInto<&'c OT>,
-    <&'c OpType as TryInto<&'c OT>>::Error: std::fmt::Debug,
-    OT: 'c,
+    for<'a> &'a OpType: TryInto<&'a OT>,
 {
     type Target = OT;
 
     fn deref(&self) -> &Self::Target {
-        self.get()
+        self.as_ref()
     }
 }
 
-impl<'c, OT, H> Clone for FatNode<'c, OT, H> {
+impl<OT, H> Copy for FatNode<'_, OT, H> {}
```

Comment:
Good catch 👍 
The `#derive` bounds are annoying when you're just wrapping references / phantom types.